designs/carbon-aware.md

```diff
@@ -0,0 +1,197 @@
+
+Approaches to handle this:
+1. Estimate extremely high emissions to effectively filter out unknown instance types (recommended)
+2. Estimate zero emissions
+
+### Launch strategy
+To enable emission based priotization, the launch strategy should be changed from `lowest-price` to `prioritized`.
+
+### Changes to consolidation (karpenter-core)
+Single Machine Consolidation (`singlemachineconsolidation.go`) and Multi Machine Consolidation (`multimachineconsolidation.go`) as well as `consolidation.go` is currently consolidating nodes to reduce costs. We want to change this when Carbon Efficient is enabled. They should consolidate to minimize carbon emissions. 
+
+### Changes to Provisioning
+Currently, provisioning (roughly) filter instances based on requirements, sort instances by price, and launch the cheapest instance. We want to change this when Carbon Efficient is enabled. It should sort instances by carbon emissions and launch the instance which has the lowest Global Warming Potential[^1].
+
+### Option 1: Use Carbon Efficient provisioning and consolidation methods
+
+#### Consolidation
+Create two new consolidation methods `carbonefficientsinglemachineconsolidation.go` and `carbonefficientmultimachineconsolidation.go` that will be used when Carbon Efficient is enabled.
+
+<details>
+
+<summary>Change to `karpenter-core/pkg/controllers/deprovisioning/controller.go`</summary>
+
+```diff
+-func NewController(clk clock.Clock, kubeClient client.Client, provisioner *provisioning.Provisioner,
+-	cp cloudprovider.CloudProvider, recorder events.Recorder, cluster *state.Cluster) *Controller {
++func NewController(ctx context.Context, clk clock.Clock, kubeClient client.Client, provisioner *provisioning.Provisioner,
++	cp cloudprovider.CloudProvider, recorder events.Recorder, cluster *state.Cluster) *Controller {
+
++	if settings.FromContext(ctx).CarbonEfficientEnabled {
++		return &Controller{
++			clock:         clk,
++			kubeClient:    kubeClient,
++			cluster:       cluster,
++			provisioner:   provisioner,
++			recorder:      recorder,
++			cloudProvider: cp,
++			lastRun:       map[string]time.Time{},
++			deprovisioners: []Deprovisioner{
++				NewExpiration(clk, kubeClient, cluster, provisioner, recorder),
++				NewDrift(kubeClient, cluster, provisioner, recorder),
++				NewEmptiness(clk),
++				NewEmptyMachineConsolidation(clk, cluster, kubeClient, provisioner, cp, recorder),
++				NewCarbonEfficientMultiMachineConsolidation(clk, cluster, kubeClient, provisioner, cp, recorder),
++				NewCarbonEfficientSingleMachineConsolidation(clk, cluster, kubeClient, provisioner, cp, recorder),
++			},
++		}
++	}
```

Comment:
Code gets outdated in a design doc, this bit already is outdated with the karpenter-core reference.